subgraph-js/src/types.ts

```diff
@@ -95,8 +96,9 @@ export function entitiesResolver({
       }
     }
 
+    const extensions = type.extensions as ApolloGraphQLObjectTypeExtensions;
     const resolveReference =
-      type.extensions.apollo?.subgraph?.resolveReference ??
+      extensions.apollo?.subgraph?.resolveReference ??
       function defaultResolveReference() {
         return reference;
       };
```

Comment:
There's no reason to name the function if it's only declared in one place.

```suggestion
      extensions.apollo?.subgraph?.resolveReference ??
      (() => reference);
```